apps/studio/src/server/modules/resource/resource.router.ts

```diff
@@ -26,6 +26,8 @@ export const resourceRouter = router({
       return resource
     }),
 
+  // TODO: Hint to the frontend that this can never return
+  // a `RootPage`
```

Comment:
you can narrow type inside kysely explicitly, but note that it's basically a cast. We should add a where clause to double check there is no rootpage:

```ts
let query = db
        .selectFrom("Resource")
        .select(["title", "permalink", "id", "type"])
        .where("Resource.type", "!=", ResourceType.RootPage) // <-- optional. ensures type narrowing is correct
        .$narrowType<{
          type: Extract<
            "Folder" | "Page" | "Collection" | "CollectionPage",
            ResourceType
          >
        }>()
        .orderBy("type", "asc")
        .orderBy("title", "asc")
```

Reply:
thanks for this, i think it makes sense in conjunction with the previuous change to filter and omit `RootPage`